contentcuration/contentcuration/models.py

```diff
@@ -1720,6 +1720,78 @@ class File(models.Model):
 
     objects = CustomManager()
 
+    _edit_filter = Q()
+    for tree_name in CHANNEL_TREES:
+        _edit_filter |= Q(
+            **{
+                "editable_channels__{}__tree_id".format(tree_name): OuterRef(
+                    "contentnode__tree_id"
+                )
+            }
+        )
+        _edit_filter |= Q(
+            **{
+                "editable_channels__{}__tree_id".format(tree_name): OuterRef(
+                    "assessment_item__contentnode__tree_id"
+                )
+            }
+        )
+
+    _view_filter = Q()
+    for tree_name in CHANNEL_TREES:
+        _view_filter |= Q(
+            **{
+                "view_only_channels__{}__tree_id".format(tree_name): OuterRef(
+                    "contentnode__tree_id"
+                )
+            }
+        )
+        _view_filter |= Q(
+            **{
+                "view_only_channels__{}__tree_id".format(tree_name): OuterRef(
+                    "assessment_item__contentnode__tree_id"
+                )
+            }
+        )
+
+    @classmethod
+    def filter_edit_queryset(cls, queryset, user):
+        user_id = not user.is_anonymous() and user.id
+        user_queryset = User.objects.filter(id=user_id)
+
+        queryset = queryset.annotate(
+            edit=Exists(user_queryset.filter(cls._edit_filter)),
```

Comment:
I believe I have noticed that Django still adds all select columns, which adds unneeded query bloat for the `EXISTS` subquery. Perhaps we can add a `.values('id')` here if it is?